index.js

```diff
@@ -55,8 +55,24 @@ function mount(app, options) {
             index(router, options.index);
         }
 
-        if (typeof options.directory === 'string') {
-            options.directory = resolve(options.basedir, options.directory);
+        if ((options.directory && typeof options.directory === 'object' && typeof options.directory.path === 'string')
+            // Deprecated: Keep this one for backwards compatibility.
+            || typeof options.directory === 'string') {
+
+            // Cast legacy config into new config.
+            if (typeof options.directory === 'string') {
+                console.error('express-enrouten:\n' +
```

Comment:
It's completely optional.
No idea if you guys want to support both methods of config. I can remove as needed.